Path: news.gmane.io!.POSTED.blaine.gmane.org!not-for-mail
From: "Guillaume Champagne" <champagne.guillaume.c-Re5JQEeQqe8AvxtiuMwx3w@public.gmane.org>
Newsgroups: gmane.comp.handhelds.openembedded.core
Subject: [RFC PATCH 1/1] image-live.bbclass: order do_bootimg after do_rootfs
Date: Wed, 26 May 2021 14:32:50 -0400
Approved: news@gmane.org
Message-ID: <20210526183250.27433-2-champagne.guillaume.c@gmail.com>
References: <20210526183250.27433-1-champagne.guillaume.c@gmail.com>
Mime-Version: 1.0
Content-Type: multipart/mixed; boundary="wuoxhxqNLP9BcJDqzrB8"
Injection-Info: ciao.gmane.io; posting-host="blaine.gmane.org:116.202.254.214";
	logging-data="6196"; mail-complaints-to="usenet@ciao.gmane.io"
Cc: Guillaume Champagne <champagne.guillaume.c-Re5JQEeQqe8AvxtiuMwx3w@public.gmane.org>
To: openembedded-core-ZwoEplunGu3dfDuKDZ/zN51Ccm5ICvs9@public.gmane.org,
	fntoth-Re5JQEeQqe8AvxtiuMwx3w@public.gmane.org
Original-X-From: bounce+67612+152277+4454657+8023386-ZwoEplunGu3dfDuKDZ/zN51Ccm5ICvs9@public.gmane.org Wed May 26 20:33:48 2021
Return-path: <bounce+67612+152277+4454657+8023386-ZwoEplunGu3dfDuKDZ/zN51Ccm5ICvs9@public.gmane.org>
Envelope-to: gchoc-openembedded-core@m.gmane-mx.org
Original-Received: from mail02.groups.io ([66.175.222.108])
	by ciao.gmane.io with esmtps (TLS1.3:ECDHE_RSA_AES_128_GCM_SHA256:128)
	(Exim 4.92)
	(envelope-from <bounce+67612+152277+4454657+8023386-ZwoEplunGu3dfDuKDZ/zN51Ccm5ICvs9@public.gmane.org>)
	id 1llyLj-0001Ly-1d
	for gchoc-openembedded-core@m.gmane-mx.org; Wed, 26 May 2021 20:33:47 +0200
X-Received: by 127.0.0.2 with SMTP id FoymYY4455992xqY0aRn3B8Y; Wed, 26 May 2021 11:33:45 -0700
X-Received: from mail-qk1-f170.google.com (mail-qk1-f170.google.com [209.85.222.170])
 by mx.groups.io with SMTP id smtpd.web09.1463.1622054023904622842
 for <openembedded-core-ZwoEplunGu3dfDuKDZ/zN51Ccm5ICvs9@public.gmane.org>;
 Wed, 26 May 2021 11:33:44 -0700
X-Received: by mail-qk1-f170.google.com with SMTP id v8so2041556qkv.1
        for <openembedded-core-ZwoEplunGu3dfDuKDZ/zN51Ccm5ICvs9@public.gmane.org>; Wed, 26 May 2021 11:33:43 -0700 (PDT)
X-Gm-Message-State: 3N9bHxdAOKB8ngtQB8JBKuB3x4454657AA=
X-Google-Smtp-Source: ABdhPJwdd/KMIrroYvzMAuSAQa/r1/DQYeOaCIhoU97vcmh09uVKYHe3C+j+LIUtr6nXaf2i68ecYg==
X-Received: by 2002:a05:620a:165a:: with SMTP id c26mr43040898qko.259.1622054023003;
        Wed, 26 May 2021 11:33:43 -0700 (PDT)
X-Received: from gchdesk.Sonatest.net (ipagstaticip-d73c7528-4de5-0861-800b-03d8b15e3869.sdsl.bell.ca. [174.94.156.236])
        by smtp.googlemail.com with ESMTPSA id h12sm2022688qkj.52.2021.05.26.11.33.42
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Wed, 26 May 2021 11:33:42 -0700 (PDT)
In-Reply-To: <20210526183250.27433-1-champagne.guillaume.c-Re5JQEeQqe8AvxtiuMwx3w@public.gmane.org>
Precedence: Bulk
List-Unsubscribe: <mailto:openembedded-core+unsubscribe-ZwoEplunGu3dfDuKDZ/zN51Ccm5ICvs9@public.gmane.org>
List-Subscribe: <mailto:openembedded-core+subscribe-ZwoEplunGu3dfDuKDZ/zN51Ccm5ICvs9@public.gmane.org>
List-Help: <mailto:openembedded-core+help-ZwoEplunGu3dfDuKDZ/zN51Ccm5ICvs9@public.gmane.org>
Original-Sender: openembedded-core-ZwoEplunGu3dfDuKDZ/zN51Ccm5ICvs9@public.gmane.org
List-Id: <openembedded-core.lists.openembedded.org>
Mailing-List: list openembedded-core-ZwoEplunGu3dfDuKDZ/zN51Ccm5ICvs9@public.gmane.org; contact openembedded-core+owner-ZwoEplunGu3dfDuKDZ/zN51Ccm5ICvs9@public.gmane.org
Delivered-To: mailing list openembedded-core-ZwoEplunGu3dfDuKDZ/zN51Ccm5ICvs9@public.gmane.org
List-Post: <mailto:openembedded-core-ZwoEplunGu3dfDuKDZ/zN51Ccm5ICvs9@public.gmane.org>
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple;
 d=lists.openembedded.org; q=dns/txt; s=20140610; t=1622054025;
 bh=96MW1qAyvvGKvqvyCsGyrIepyPX05C3xyBCemzw8Iic=;
 h=Cc:Content-Type:Date:From:Subject:To;
 b=nWd2yxH3LyeED9mr/lXvJEy+ulbfTMKLBhjrlgKC+pScoK4EKWeYcl9I6gv1PjyYd5O
 mhaNCj7nhYJNYvnO0snO8+19lC05U459DJH7spiJ+6umwEm9RafCBmbGPaVHii+h1wuo8
 3d4k3VzxAlEjkuTe+Q+FPM1CcGcZHjdOFKQ=
Xref: news.gmane.io gmane.comp.handhelds.openembedded.core:150497
Archived-At: <http://permalink.gmane.org/gmane.comp.handhelds.openembedded.core/150497>

--wuoxhxqNLP9BcJDqzrB8
Content-Transfer-Encoding: 8bit

do_bootimg expects IMGDEPLOYDIR to exist, since it stores its artifacts
there. Therefore, do_bootimg should run after do_rootfs because
IMGDEPLOYDIR is created before do_rootfs runs since IMGDEPLOYDIR is
contained in do_rootfs' [cleandirs] varflag.

When do_bootimg depends on ${PN}:do_image_${LIVE_ROOTFS_TYPE},
do_bootimg is correctly ordered after do_rootfs because
do_image_${FSTYPE} tasks are added after do_image and do_image itself is
added after do_rootfs.

However, when do_bootimg doesn't depend on
${PN}:do_image_${LIVE_ROOTFS_TYPE}
(introduced by: 96f47c39f1d17f073243913d524bde84add41d8f), do_bootimg
can run before do_rootfs, thus before IMGDEPLOYDIR is created. To
avoid this situation, do_bootimg is now explicitly ordered after
do_rootfs.

Signed-off-by: Guillaume Champagne <champagne.guillaume.c-Re5JQEeQqe8AvxtiuMwx3w@public.gmane.org>
---
 meta/classes/image-live.bbclass | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meta/classes/image-live.bbclass b/meta/classes/image-live.bbclass
index 8b08305cdb..fd876ed8e1 100644
--- a/meta/classes/image-live.bbclass
+++ b/meta/classes/image-live.bbclass
@@ -261,4 +261,4 @@ python do_bootimg() {
 do_bootimg[subimages] = "hddimg iso"
 do_bootimg[imgsuffix] = "."
 
-addtask bootimg before do_image_complete
+addtask bootimg before do_image_complete after do_rootfs
-- 
2.20.1


--wuoxhxqNLP9BcJDqzrB8
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: quoted-printable
Content-Disposition: inline


-=3D-=3D-=3D-=3D-=3D-=3D-=3D-=3D-=3D-=3D-=3D-
Links: You receive all messages sent to this group.
View/Reply Online (#152277): https://lists.openembedded.org/g/openembedded=
-core/message/152277
Mute This Topic: https://lists.openembedded.org/mt/83107841/4454657
Group Owner: openembedded-core+owner-ZwoEplunGu3dfDuKDZ/zN51Ccm5ICvs9@public.gmane.org
Unsubscribe: https://lists.openembedded.org/g/openembedded-core/unsub [gch=
oc-openembedded-core@m.gmane-mx.org]
-=3D-=3D-=3D-=3D-=3D-=3D-=3D-=3D-=3D-=3D-=3D-


--wuoxhxqNLP9BcJDqzrB8--

